saml.php

```diff
@@ -0,0 +1,95 @@
+<?php
+/* Copyright (c) 1998-2017 ILIAS open source, Extended GPL, see docs/LICENSE */
+
+/**
+ * @author Michael Jansen <mjansen@databay.de>
+ */
+
+require_once 'libs/composer/vendor/autoload.php';
+$as = new SimpleSAML_Auth_Simple('default-sp');
+
+if(isset($_GET['action']) && $_GET['action'] == 'logout' && isset($_GET['logout_url']) && strlen($_GET['logout_url']))
+{
+	$as->logout(array(
+		'ReturnTo'         => $_GET['logout_url'],
+		'ReturnStateParam' => 'LogoutState',
+		'ReturnStateStage' => 'ilLogoutState'
+	));
+}
+
+$saml_config = SimpleSAML_Configuration::getInstance();
+$store_type  = $saml_config->getString('store.type', false);
+
+$session = SimpleSAML_Session::getSessionFromRequest();
+
+if(isset($_GET['target']) && !isset($_GET['returnTo']))
+{
+	$_GET['returnTo'] = $_GET['target'];
+}
+
+if(isset($_GET['returnTo']))
+{
+	$session->setData('example:set_target', 'il_target', $_GET['returnTo']);
+}
+else
+{
+	$session->deleteData('example:set_target', 'il_target');
+}
+
+$as->requireAuth();
+
```

Comment:
This is where the actual authentication at some predefined Idp takes place. Usually this Idp is listed in the config/authsources.php file in the simplesaml-directory. The corresponding Idp-metadata, as for instance login and logout -urls and the certificate, is then loaded from somewhere, usually from the configuration file metadata/saml20-idp-remote.php, again, within the simplesaml directory. Since i can not pinpoint any location, where this procedure is substituted by some different procedure, i assume that this is case here also, meaning one would have to locate the corresponding files manually in the places mentioned above at installation?

A word of warning: 
It would be, i.m.o., problematic to keep the configuration folders within the webfolder. If someone resorts to to an xml-configuration file this could be accessed by someone simply bu guessing the correct path. However, this would be considered a configuration mistake. Of course, one may specify a custom metadata directory somehere outside the webfolder in the config.php file using the metadatadir option. In general it is recomended to keep the whole simplesamlphp folder outside the webdir and only link simplesamlphp/www into the proper position within.
A much cleaner solution would probably be to store all the metadata within the db and make it configurable in some ilias-based GUI. The metadata might be saved, for instance, using ilSamlIdp.